src/market/market.module.ts

```diff
@@ -189,49 +205,41 @@ export class MarketModuleImpl implements MarketModule {
     this.fileServer = deps.fileServer;
   }
 
-  async buildDemand(options: DemandOptionsNew, payerDetails: PayerDetails): Promise<DemandSpecification> {
-    const demandSpecificConfig = new DemandConfig(options);
-    const builder = new DecorationsBuilder();
-
-    // Apply additional modifications
-    const pkgOptions = await this.applyLocalGVMIServeSupport(options);
-    const taskDecorations = await Package.create(pkgOptions).getDemandDecoration();
-
-    builder.addDecorations([taskDecorations]);
-
-    // Configure basic properties
-    builder
-      .addProperty("golem.srv.caps.multi-activity", true)
-      .addProperty("golem.srv.comp.expiration", Date.now() + demandSpecificConfig.expirationSec * 1000)
-      .addProperty("golem.node.debug.subnet", demandSpecificConfig.subnetTag)
-      .addProperty("golem.com.payment.debit-notes.accept-timeout?", demandSpecificConfig.debitNotesAcceptanceTimeoutSec)
-      .addConstraint("golem.com.pricing.model", "linear")
-      .addConstraint("golem.node.debug.subnet", demandSpecificConfig.subnetTag);
-
-    // Configure mid-agreement payments
-    builder
-      .addProperty(
-        "golem.com.scheme.payu.debit-note.interval-sec?",
-        demandSpecificConfig.midAgreementDebitNoteIntervalSec,
-      )
-      .addProperty("golem.com.scheme.payu.payment-timeout-sec?", demandSpecificConfig.midAgreementPaymentTimeoutSec);
-
-    // Configure payment platform
-    builder
-      .addProperty(`golem.com.payment.platform.${payerDetails.getPaymentPlatform()}.address`, payerDetails.address)
-      .addConstraint(`golem.com.payment.platform.${payerDetails.getPaymentPlatform()}.address`, "*")
-      .addProperty("golem.com.payment.protocol.version", "2")
-      .addConstraint("golem.com.payment.protocol.version", "1", ComparisonOperator.Gt);
-
-    return builder.getDemandSpecification(payerDetails.getPaymentPlatform(), demandSpecificConfig.expirationSec);
+  async buildDemandDetails(options: BuildDemandOptions, payerDetails: PayerDetails): Promise<DemandSpecification> {
+    const builder = new DemandBodyBuilder();
+
+    // Instruct the builder what's required
+    const basicConfig = new BasicDemandDirectorConfig(options.basic);
+    const basicDirector = new BasicDemandDirector(basicConfig);
+    basicDirector.apply(builder);
+
+    const workloadOptions = options.activity
```

Comment:
I am considering the name... so we use the name workload here, whereas in the options we use the name activity... I think I like the name `workload` better than `activity` in this context.. but this is just a suggestion